it/test/uk/gov/hmrc/timetopayproxy/controllers/TimeToPayProxyControllerItSpec.scala

```diff
@@ -0,0 +1,319 @@
+
+              val response: WSResponse = await(
+                requestForAffordableQuotes.post(Json.toJson(affordableQuoteRequest))
+              )
+
+              val expectedTtppErrorResponse: TtppErrorResponse =
+                TtppErrorResponse(statusCode = 400, errorMessage = "An unknown error has occurred")
+
+              response.json shouldBe Json.toJson(expectedTtppErrorResponse)
+              response.status shouldBe 400
+            }
+
+            "with one item in the list of 'failures'" in new TimeToPayProxyControllerTestBase {
+              val timeToPayError: TimeToPayError =
+                TimeToPayError(failures = Seq(Error(code = "BAD_REQUEST", reason = "only reason")))
+
+              stubPostWithResponseBody(url = "/auth/authorise", status = 200, responseBody = "null")
+              stubPostWithResponseBody(
+                url = "/debts/time-to-pay/affordability/affordable-quotes",
+                status = 400,
+                responseBody = Json.toJson(timeToPayError).toString()
+              )
+
+              val requestForAffordableQuotes: WSRequest = buildRequest("/self-serve/affordable-quotes")
+
+              val response: WSResponse = await(
+                requestForAffordableQuotes.post(Json.toJson(affordableQuoteRequest))
+              )
+
+              val expectedTtppErrorResponse: TtppErrorResponse =
+                TtppErrorResponse(statusCode = 400, errorMessage = "only reason")
+
+              response.json shouldBe Json.toJson(expectedTtppErrorResponse)
+              response.status shouldBe 400
+            }
+
+            "with multiple items in the list of 'failures', TTP-Proxy should return the first message" in new TimeToPayProxyControllerTestBase {
+              val timeToPayError: TimeToPayError =
+                TimeToPayError(failures =
+                  Seq(
+                    Error(code = "BAD_REQUEST", reason = "first reason"),
+                    Error(code = "SERVICE_UNAVAILABLE", reason = "second reason"),
+                    Error(code = "INTERNAL_SERVER_ERROR", reason = "third reason")
+                  )
+                )
+
+              stubPostWithResponseBody(url = "/auth/authorise", status = 200, responseBody = "null")
+              stubPostWithResponseBody(
+                url = "/debts/time-to-pay/affordability/affordable-quotes",
+                status = 400,
+                responseBody = Json.toJson(timeToPayError).toString()
+              )
+
+              val requestForAffordableQuotes: WSRequest = buildRequest("/self-serve/affordable-quotes")
+
+              val response: WSResponse = await(
+                requestForAffordableQuotes.post(Json.toJson(affordableQuoteRequest))
+              )
+
+              val expectedTtppErrorResponse: TtppErrorResponse =
+                TtppErrorResponse(statusCode = 400, errorMessage = "first reason")
```

Comment:
I've thought more about it. Its not important to take all the possible error messages and send them through. Just the first will be fine for TTP-proxy. Lmk what you think and mark this as resolved if you're happy

Reply:
I'm happy with either because we get different (but similar outcomes):
- If we see all the error messages = visual crowding, but we know for sure the validation picks up on the rest
- If we don't see all the error messages = simpler to read + identify, leaves rest of the validation in the grey area (but if we have enough unit/integration tests, this is taken care of, which it is here)